src/engine/sim_engine/blockdev.rs

```diff
@@ -19,13 +22,40 @@ use super::randomization::Randomizer;
 pub struct SimDev {
     pub devnode: PathBuf,
     rdm: Rc<RefCell<Randomizer>>,
-    uuid: Uuid,
+    pub uuid: Uuid,
+    user_info: Option<String>,
+    hardware_info: Option<String>,
+    initialization_time: u64,
 }
 
 impl BlockDev for SimDev {
     fn devnode(&self) -> PathBuf {
         self.devnode.clone()
     }
+
+    fn user_info(&self) -> Option<&str> {
+        self.user_info.as_ref().map(|x| &**x)
+    }
+
+    fn set_user_info(&mut self, user_info: Option<&str>) -> () {
+        self.user_info = user_info.map(|x| x.to_owned());
+    }
+
+    fn hardware_info(&self) -> Option<&str> {
+        self.hardware_info.as_ref().map(|x| &**x)
+    }
+
+    fn initialization_time(&self) -> DateTime<Utc> {
```

Comment:
Probably ```initialization_time``` result should just be u64. Otherwise, it seems like there's some pointless round-tripping to take a date time, change into a timestamp, and then retrieve it and turn it back into a date time.

Reply:
I think the type is useful to understand what it is when reading the code, and converting to a standard format internally ensures we don't have hidden dependencies between our on-disk format and our API.